src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -375,6 +382,8 @@ func (a *StorageMinerActorCode_I) RecoverFaults(rt Runtime, recoveringSet sector
 
 			// SendMessage(sma.PublishStorageDeals) or sma.ResumeStorageDeals?
 			// throw if miner cannot cover StorageDealCollateral
+			// TODO: EnsureDealCollateral
+			panic("TODO")
```

Comment:
Note: dealing with dynamic collateral sort of works for us here. We could add a call to SPA to `tryLockCollateral(expectedPower)` that locks up collateral for the expected power after recovery, in advance of marking the sectors recovered. If the call fails, refuse to recover the sectors until the miner deposits enough that the tryLock succeeds. For a follow-up

Reply:
ah, this is something else, pledge collateral is checked at PoSt submission. This is on deal collateral which is a bit more complicated to account for..

Reply:
good news, we actually dont need `EnsureDealCollateral` here because storage deal collateral is not slashed at Declared/Detected Faults. It is only slashed in TerminatedFaults and all will be irrecoverably slashed and hence we dont need a top up. cc @sternhenri @jzimmerman